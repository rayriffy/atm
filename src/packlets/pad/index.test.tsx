import { render, act } from '@testing-library/react'
import { describe, it, expect, vi, beforeEach } from 'vitest'

import { Pad } from './'

const propsSpy = vi.fn()
vi.mock('./button', () => ({
  PadButton: (props: unknown) => {
    propsSpy(props)
    return <div>PadButton</div>
  },
}))

describe('<Pad />', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = render(<Pad />)
    expect(container).toMatchSnapshot()
  })

  it('should render 11 buttons (10 number + 1 del)', () => {
    render(<Pad />)
    expect(propsSpy).toBeCalledTimes(11)
  })

  it('should be able to click a button', () => {
    const onClickSpy = vi.fn()
    render(<Pad onClick={onClickSpy} />)

    expect(onClickSpy).not.toBeCalled()

    act(() => {
      propsSpy.mock.calls[0][0].onClick()
    })

    expect(onClickSpy).toBeCalledTimes(1)
    expect(onClickSpy).toBeCalledWith('1')
  })
})
