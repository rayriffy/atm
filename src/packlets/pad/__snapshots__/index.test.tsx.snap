// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Pad /> > should match snapshot 1`] = `
<div>
  <div
    class="chakra-stack chakra-container css-gfniwp"
  >
    <div
      class="css-vw1wj9"
    >
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
      <div>
        PadButton
      </div>
    </div>
  </div>
</div>
`;
